### PR TITLE
[ICDS Dashboard]: District wise data on "Private School going Children" for month July 20 & August 20 for the state of Gujarat.

### PR DESCRIPTION
**Pull request by ShubhamBansal1997**
District wise data on "Private School going Children" for month July 20 & August 20 for the state of Gujarat.
https://app.asana.com/0/1187146330310233/1187439890927105/f

### CODE REVIEW

File: custom/icds_reports/management/commands/data_pull_private.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        """
+
+        months = [datetime.date(2020, 7, 1), datetime.date(2020, 8, 1)]
+        case_accessor = CaseAccessors('icds-cas')
+        get_district_ids = self.get_district_ids_dict()
+        for month in months:
+            excel_data = [['district', 'count']]
+            person_case_ids = {}
+            for district_id in get_district_ids.keys():
+                person_case_ids[district_id] = []
```
Comment (Rohit25negi) [inline custom/icds_reports/management/commands/data_pull_private.py:50]: can you use the [defaultdict](https://www.geeksforgeeks.org/defaultdict-in-python/) here?

File: custom/icds_reports/management/commands/data_pull_private.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        for loc in locations:
+            district_names[loc['district_id']] = loc['district_name']
+        return district_names
+
+    def handle(self, **options):
+        query = """
+            SELECT "awc_location_months"."district_id", "child_health_monthly"."child_person_case_id" FROM "public"."child_health_monthly" "child_health_monthly"
+            LEFT JOIN "public"."awc_location_months" "awc_location_months" ON (
+                ("awc_location_months"."month" = "child_health_monthly"."month") AND
+                ("awc_location_months"."awc_id" = "child_health_monthly"."awc_id")
```
Comment (Rohit25negi) [inline custom/icds_reports/management/commands/data_pull_private.py:39]: does it needs to compare supervisor_id as well? as both are distributed relations

**Review by Rohit25negi — COMMENTED** [review]
left two comments

**Review by Rohit25negi — APPROVED** [review]

File: custom/icds_reports/management/commands/data_pull_private.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                ("awc_location"."supervisor_id" = "child_health_monthly"."supervisor_id")
+            ) LEFT OUTER JOIN "public"."icds_dashboard_migration_forms" "agg_migration" ON (
+                ("child_health_monthly"."child_person_case_id" = "agg_migration"."person_case_id") AND
+                ("agg_migration"."month" = '{month}') AND
+                ("child_health"."state_id" = "agg_migration"."state_id") AND
+                ("child_health"."supervisor_id" = "agg_migration"."supervisor_id")
+            ) WHERE "child_health_monthly".month='{month}' AND "awc_location".state_id='{state_id}'
+            AND "child_health_monthly"."open_in_month"=1 AND "child_health_monthly"."alive_in_month"=1
+            AND "child_health_monthly"."age_tranche" IN ('48', '60', '72')
+            AND ("agg_migration"."is_migrated" IS DISTINCT FROM 1 OR "agg_migration"."migration_date"::date >= '{month}');
```
[the remark this answers is not in the document]
Comment (Rohit25negi) [reply custom/icds_reports/management/commands/data_pull_private.py:49]: can you use `valid_all_registered_in_month` column directly from chm table that way you wont need this join to check migration status?

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
left one comment.

File: custom/icds_reports/management/commands/data_pull_private.py (hunk 2 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/management/commands/data_pull_private.py]: updated!

File: custom/icds_reports/management/commands/data_pull_private.py (hunk 1 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/management/commands/data_pull_private.py]: updated!

File: custom/icds_reports/management/commands/data_pull_private.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ShubhamBansal1997) [reply custom/icds_reports/management/commands/data_pull_private.py]: updated!

**Review by Rohit25negi — APPROVED** [review]

**Pull request closed without merging**